src/Interface/Application/ModuleWidget.h

```diff
@@ -301,13 +307,19 @@ private Q_SLOTS:
   void addOutputPortsToLayout(int index);
   void addOutputPortsToWidget(int index);
   void removeOutputPortsFromWidget(int index);
+  void updateProgrammablePorts();
   QHBoxLayout* inputPortLayout_;
   QHBoxLayout* outputPortLayout_;
   bool deleting_;
   static bool networkBeingCleared_;
   const QString defaultBackgroundColor_;
   bool isViewScene_; //TODO: lots of special logic around this case.
 
+  typedef boost::bimap<QString, int> ColorStateLookup;
```

Comment:
I'm really curious about the bimap and why you chose to use a bimap over `std::unordered_map` or `std::map`?

Reply:
There is lookup in both directions--otherwise I'd just use a regular map